src/inc/quic_datapath.h

```diff
@@ -107,9 +107,13 @@ PacketSizeFromUdpPayloadSize(
     _In_ uint16_t UdpPayloadSize
     )
 {
-    return Family == QUIC_ADDRESS_FAMILY_INET ?
+    uint32_t PayloadSize = Family == QUIC_ADDRESS_FAMILY_INET ?
         UdpPayloadSize + CXPLAT_MIN_IPV4_HEADER_SIZE + CXPLAT_UDP_HEADER_SIZE :
         UdpPayloadSize + CXPLAT_MIN_IPV6_HEADER_SIZE + CXPLAT_UDP_HEADER_SIZE;
+    if (PayloadSize > UINT16_MAX) {
+        PayloadSize = UINT16_MAX;
+    }
+    return (uint16_t)PayloadSize;
```

Comment:
Do we need to do this here? Can we handle this at the (hopefully only 1) caller that is passing in too big of a value? Probably update the annotation here too to indicate you're not supposed to pass too big of a value?

Reply:
Its only 1 place that calls it with potentially too large of value that we would need to truncate. However, moving the logic outside of here would increase the complication because of the different families.

Reply:
Hmmm. One thing we've talked about is changing all our "packet size" stuff to be in terms of UDP payload. Perhaps that'd help this? (for now, just keep as is I guess)

Reply:
The specific thing that makes this an issue is that the TP for max peer mtu has a max of 65527. And a lot of the other implementations set to that. And thats the value that overflows. Moving to udp payload size here wouldnt actually fix the bug, as I want MTU.